server/src/main/java/com/cloud/hypervisor/HypervisorGuruBase.java

```diff
@@ -240,7 +239,7 @@ protected VirtualMachineTO toVirtualMachineTO(VirtualMachineProfile vmProfile) {
         // Workaround to make sure the TO has the UUID we need for Niciri integration
         VMInstanceVO vmInstance = _virtualMachineDao.findById(to.getId());
         // check if XStools/VMWare tools are present in the VM and dynamic scaling feature is enabled (per zone/global)
-        Boolean isDynamicallyScalable = vmInstance.isDynamicallyScalable() && UserVmManager.EnableDynamicallyScaleVm.valueIn(vm.getDataCenterId());
+        Boolean isDynamicallyScalable = vmInstance.isDynamicallyScalable();
```

Comment:
where this global setting value "_UserVmManager.EnableDynamicallyScaleVm.valueIn(vm.getDataCenterId())_" is being checked now ? Please update the above comment appropriately.

Reply:
During VM deployment when VM entry is persisted, VM's dynamic scalability is evaluated based on all options user input, template, service offering and global setting. So no need to check with global setting while preparing the transfer object.
Removing the comment.